mono/metadata/threads.c

```diff
@@ -338,6 +341,8 @@ mono_thread_set_interruption_requested (MonoInternalThread *thread)
 			new_state = old_state | INTERRUPT_ASYNC_REQUESTED_BIT;
 	} while (InterlockedCompareExchangePointer ((volatile gpointer)&thread->thread_state, (gpointer)new_state, (gpointer)old_state) != (gpointer)old_state);
 
+	InterlockedIncrement (&thread_interruption_requested);
```

Comment:
We should only bump it if we expect that the thread will be able to serve the interruption reasonably fast. There's a significant slowdown associated with having thread_int_req > 0.